airflow/models/taskinstance.py

```diff
@@ -1009,19 +1029,27 @@ def signal_handler(signum, frame):
                     self.log.error("Failed when executing execute callback")
                     self.log.exception(e3)
 
-                # If a timeout is specified for the task, make it fail
-                # if it goes beyond
-                result = None
-                if task_copy.execution_timeout:
-                    try:
-                        with timeout(int(
-                                task_copy.execution_timeout.total_seconds())):
+                with ExitStack() as exit_stack:
+                    from airflow.task.context.current import set_current_context
+                    exit_stack.enter_context(set_current_context(context))
+                    user_defined_exec_context = get_user_defined_execute_context()
+                    if user_defined_exec_context is not None:
+                        try:
+                            exit_stack.enter_context(user_defined_exec_context)
```

Comment:
I am not sure if this try catch for user code is required (minor)

Reply:
Exception received without try/except:
`TypeError: incorrect_user_defined_context_func() takes 0 positional arguments but 2 were given`

I think you are correct, the existing exception is good enough